### PR TITLE
Added flatten variants that operate on SequenceTypes

### LINKED ISSUE #2956: Add flatten variants that operate on SequenceTypes
This would help with otherwise ambiguous uses of `flatMap` like [this](https://github.com/Carthage/Carthage/pull/1329/files#r65486309).

You should be able to write something like this:

```swift
SignalProducer<[Int], NoError>(value: [1, 2, 3]).flatten(.Concat)
```

### PR DESCRIPTION
**Pull request by miikegb**
This PR attempts to fix #2956

### CODE REVIEW

**Comment by RuiAAPeres** [comment]
You are extending and constraining on a `SequenceType`, but on the tests you describe it as a `CollectionType`:
```swift
+			it("works with CollectionType as a value") {
```

**Comment by miikegb** [comment]
Good catch, I meant `SequenceType`. Will submit a fix.

File: ReactiveCocoaTests/Swift/FlattenSpec.swift [hunk 1]
```diff
@@ -248,6 +248,20 @@ class FlattenSpec: QuickSpec {
...
+			it("works with SequenceType as a value") {
+				let (signal, innerObserver) = Signal<[Int], NoError>.pipe()
+				let sequence = [1, 2, 3]
+				var observedValues = [Int]()
+				
+				signal
+					.flatten(.Concat)
+					.observeNext { value in
+						observedValues.append(value)
+				}
```
Comment (mdiep) [inline ReactiveCocoaTests/Swift/FlattenSpec.swift]: The indentation of this `}` is off a level.

File: ReactiveCocoaTests/Swift/FlattenSpec.swift [hunk 2]
```diff
@@ -402,6 +416,20 @@ class FlattenSpec: QuickSpec {
...
+			it("works with SequenceType as a value") {
+				let sequence = [1, 2, 3]
+				var observedValues = [Int]()
+				
+				let producer = SignalProducer<[Int], NoError>(value: sequence)
+				producer
+					.flatten(.Latest)
+					.startWithNext { value in
+						observedValues.append(value)
+				}
```
Comment (mdiep) [inline ReactiveCocoaTests/Swift/FlattenSpec.swift]: Here too.

**Comment by mdiep** [comment]
I didn't see that changes were made to this. This is great! We'll get this in the release for Swift 2.3 / Xcode 8 support.

**Pull request merged**